Update the ptrace_scope if clause to check the correct string and add a check in the kitchen test for the config file chef generates

dreambeyondorange opened the pull request:
Fix bug in disable_ptrace action to allow sysctl to disable ptrace.

Previously the setting for Ubuntu systems was not getting used due to a misalignment between the cookbook and cli.
On the cli, node['cluster']['enable_efa'] was set to 'efa' in the dna.json file, but the cookbook assumed the parameter was set to 'compute'.  This causes the disable_ptrace action if clause to be skipped.

This page describes the use of sysctl https://manpages.ubuntu.com/manpages/bionic/man5/sysctl.conf.5.html
This page describes the function of the ptrace_scope parameter https://www.kernel.org/doc/Documentation/security/Yama.txt
Chef actually generates a config file in /etc/sysctl.d/ for the parameter that is changed so it should persist. https://docs.chef.io/resources/sysctl/

Extended the efa_debian_system_settings_configured kitchen test to include the file addition but spec tests remain the same.

### Tests
* `efa_debian_system_settings_configured` executed manually and added to `tag:config*`
* Confirmed that changing the parameter `enable_efa` in the dna.json results in the resource running
```
Recipe: aws-parallelcluster-config::base
  * efa[Configure system for EFA] action configure[2023-06-15T15:05:05+00:00] INFO: Processing efa[Configure system for EFA] action configure (aws-parallelcluster-config::base line 38)

    * sysctl[kernel.yama.ptrace_scope] action apply[2023-06-15T15:05:05+00:00] INFO: Processing sysctl[kernel.yama.ptrace_scope] action apply ((eval) line 19)

      * directory[/etc/sysctl.d] action create[2023-06-15T15:05:05+00:00] INFO: Processing directory[/etc/sysctl.d] action create (/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource/sysctl.rb line 139)
 (up to date)
      * file[/etc/sysctl.d/99-chef-kernel.yama.ptrace_scope.conf] action create[2023-06-15T15:05:05+00:00] INFO: Processing file[/etc/sysctl.d/99-chef-kernel.yama.ptrace_scope.conf] action create (/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource/sysctl.rb line 141)
```

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

dreambeyondorange committed:
Fix bug in disable_ptrace action to allow sysctl to disable ptrace.

Previously the setting for Ubuntu systems was not getting used due to a misalignment between the cookbook and cli.
On the cli, node['cluster']['enable_efa'] was set to 'efa' in the dna.json file, but the cookbook assumed the
parameter was set to 'compute'.  This causes the disable_ptrace action if clause to be skipped.
This page describes the use of sysctl https://manpages.ubuntu.com/manpages/bionic/man5/sysctl.conf.5.html
This page describes the function of the ptrace_scope parameter https://www.kernel.org/doc/Documentation/security/Yama.txt
Chef actually generates a config file in /etc/sysctl.d/ for the parameter that is changed so it should persist.
https://docs.chef.io/resources/sysctl/

Extended the efa_debian_system_settings_configured kitchen test to include the file addition but spec tests remain the same.

enrico-usai commented:
Can you please:
* add a reference for the ubuntu documentation explaining this configuration step.
* execute the `efa_debian_system_settings_configured` inspec test to verify it's still passing and add the results in the pr description (let's add tag:config to this, to be sure we have it in our daily tests).
* extend the `efa_debian_system_settings_configured` inspec test to verify the file is created as expected and eventually if there is a way to see if it's applied in the system
* extend the ChefSpec `cookbooks/aws-parallelcluster-environment/spec/unit/resources/efa_spec.rb` to cover the creation of the file

dreambeyondorange commented:
> Can you please:
> 
> * add a reference for the ubuntu documentation explaining this configuration step.
> * execute the `efa_debian_system_settings_configured` inspec test to verify it's still passing and add the results in the pr description (let's add tag:config to this, to be sure we have it in our daily tests).
> * extend the `efa_debian_system_settings_configured` inspec test to verify the file is created as expected and eventually if there is a way to see if it's applied in the system
> * extend the ChefSpec `cookbooks/aws-parallelcluster-environment/spec/unit/resources/efa_spec.rb` to cover the creation of the file

I updated the description to include the test I ran.  Updated the inspec and spec tests.

demartinofra reviewed (dismissed):
LGTM. Please update the changelog before merging

dreambeyondorange commented:
skipping system tests as they were already passing before updating the changelog

enrico-usai commented on CHANGELOG.md:
```diff
@@ -22,6 +22,9 @@ This file is used to list changes made in each version of the AWS ParallelCluste
   in order to reduce bootstrap time and avoid a potential point of failure.
 
+**BUG FIXES**
+- Fix an issue that was preventing ptrace protection from being disabled on Ubuntu and allowing Cross Memory Attach (CMA) in libfabric
```
missing final dot.
